### DISCUSSION TITLE
Warning coming in console - 'Warning: Prop `style` did not match'

### DISCUSSION THREAD

**Discussion by tarun1992** [vercel/next.js#21697] [Help]
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

12

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 10

**How are you deploying your application?**

Vercel

**Describe the Bug**

The warning is coming in the console after I refresh my application - 

Although, I figured out how to resolve the issue as it is coming from the carousel library. The issue gets resolved after I use dynamic with ssr: false 'const Carousel = dynamic(() => import("react-multi-carousel"));'. But it is impacting my SEO as it then renders on the client-side and my application heavily uses carousels.

The warning - 

Warning: Prop `style` did not match. Server: "flex:1 0 33.3%;position:relative;width:auto" Client: "flex:1 0 100.0%;position:relative;width:auto"
    at li
    at CarouselItems (http://localhost:3000/_next/static/chunks/1.js?ts=1612016062593:47:273)
    at ul
    at div
    at Carousel (http://localhost:3000/_next/static/chunks/1.js?ts=1612016062593:35:1297)
    at LoadableComponent (http://localhost:3000/_next/static/chunks/pages/index.js?ts=1612016062593:1603:37)

**Expected Behavior**

There should be no warning in the console.

**To Reproduce**

 This is not coming in the production build and is occurring  via the carousel library. But, it was not coming in the 9th version and occurred after upgrading it to 10.0.6.

**Comment by studynetskill** [comment]
I also meet this problem,
This is occur in my console:
> react-dom.development.js:67 Warning: Prop `style` did not match. Server: "null" Client: "row-gap:0"
>    at div
>    at Row (http://localhost:3000/_next/static/chunks/pages/index.js?ts=1612843702599:18975:34)
>    at Home
>    at MyApp (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1612843702599:9254:24)
>    at ErrorBoundary (http://localhost:3000/_next/static/chunks/main.js?ts=1612843702599:755:47)
>   at ReactDevOverlay (http://localhost:3000/_next/static/chunks/main.js?ts=1612843702599:859:23)
>   at Container (http://localhost:3000/_next/static/chunks/main.js?ts=1612843702599:9233:5)
>    at AppContainer (http://localhost:3000/_next/static/chunks/main.js?ts=1612843702599:9721:24)
>    at Root (http://localhost:3000/_next/static/chunks/main.js?ts=1612843702599:9852:24)
It happend when i use this:
`"antd": "^4.12.2"`
`"next": "10.0.6"`
Could u tell me how to fix is? Or i have to go back to  the 9th version?

**Comment by benweier** [comment]
@tarun1992 @studynetskill This warning comes from React when it reconciles the client-side DOM with the initial server-side DOM output. The most common reason I see for client/server differences is media queries. Simply put the server (inherently) doesn't support media queries because there's no `window`.

Any component that renders different content or styles based on the breakpoint in the browser will risk this warning no matter what because, in effect, the server renders its output as though window width is 0 (it's likely that this warning doesn't display when you put your browser into [Device Mode](https://developers.google.com/web/tools/chrome-devtools/device-mode/).

There are ways to hint to the server what breakpoint the browser _may_ be in based on request headers, but it's not guaranteed. The user agent string, for example, can tell you if the user is iOS or Windows 10 (allowing for a best-guess browser size) but won't tell you if the device is landscape/portrait (not to mention all the different screen sizes that exist) or if the user has resized their browser window.

You then need to make sure the library or component is aware of this breakpoint "suggestion" by providing a fallback value when `window.matchMedia` isn't available - unfortunately many claims of "SSR Support" don't always make it easy to control a fallback value.

**Comment by trediv** [comment]
In case somebody encountered this issue. My problem was in Chrome with the Dark Reader extension enabled.  So check out all your extension or in incognito mode to discard this case.
